SpatialGDK/Source/Private/Utils/ComponentReader.cpp

```diff
@@ -103,7 +103,39 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject*
 			{
 				if (Cmd.Type == REPCMD_DynamicArray)
 				{
-					ApplyArray(ComponentObject, FieldId, Cast<UArrayProperty>(Cmd.Property), Data, SwappedCmd.Offset, Cmd.ParentIndex);
+					UArrayProperty* ArrayProperty = Cast<UArrayProperty>(Cmd.Property);
+					bool bProcessedArray = false;
+
+					if (UStructProperty* ParentStruct = Cast<UStructProperty>(Parent.Property))
+					{
+						if (ParentStruct->Struct->IsChildOf(FFastArraySerializer::StaticStruct()))
+						{
+							// Read array into a temporary array so the appropriate remove/add operations can be processed
```

Comment:
Might be worth refactoring this into a ApplyFastArray method

Reply:
I hear you, but it would require passing a quite few objects, and it's only called in this one spot, so I'm inclined to leave it like this for now. Unless you strongly object?

Reply:
I'm not particularly strong about it. It's not for re-usability, just for method size reduction. Leave as is :D